server/controllers/refer/programmeHistoryController.ts

```diff
@@ -0,0 +1,21 @@
+import type { Request, Response, TypedRequestHandler } from 'express'
+
+import type { CourseService } from '../../services'
+import { CourseUtils, TypeUtils } from '../../utils'
+
+export default class ProgrammeHistoryController {
```

Comment:
Per previous comment, I think we could be calling this `CourseParticipationsController` per the entity name